lib/backend/test/suite.go

```diff
@@ -100,12 +101,17 @@ func (s *BackendSuite) CRUD(c *check.C) {
 	c.Assert(err, check.IsNil)
 	c.Assert(string(out.Value), check.Equals, string(item.Value))
 
-	// put with binary key and value succeeds
-	key := make([]byte, 1024)
-	rand.Read(key)
+	// put with large key and binary value succeeds.
+	//  NB: DynamoDB has a maximum overall key length of 1024 bytes, so we trim
+	//      our largest keysize to match.
+	keyLen := (1024 / 2) - len(prefix(""))
```

Comment:
Given an obscure behavior change to `prefix` to generate prefixes larger than 512 bytes, this will start producing negative lengths.

Reply:
Obscure in the sense of spooky action at a distance? I.e. a change there with unintended consequences here? 

Perhaps I should just pick a largish magic number rather than try to be clever.

Reply:
Yes, basically. We have a saying in Russian that roughly translates to "code as reinforced concrete" in the sense that no side-effects anywhere won't have no effect on nothing.

Reply:
Done